src/Qir/Runtime/stdlib/src/bigints.rs

```diff
@@ -0,0 +1,335 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#![deny(clippy::all, clippy::pedantic)]
+
+use crate::update_counts;
+use num_bigint::BigInt;
+use std::{mem::ManuallyDrop, rc::Rc};
+
+#[no_mangle]
+pub extern "C" fn __quantum__rt__bigint_create_i64(input: i64) -> *const BigInt {
+    Rc::into_raw(Rc::new(input.into()))
+}
+
+/// # Safety
+///
+/// This function expects the second argument to be a well-formed C-style array of signed big-endian bytes
+/// with the size of that array passed as the first argument.
+#[no_mangle]
+pub unsafe extern "C" fn __quantum__rt__bigint_create_array(
+    size: u32,
+    input: *const u8,
+) -> *const BigInt {
+    Rc::into_raw(Rc::new(BigInt::from_signed_bytes_be(
+        std::slice::from_raw_parts(input, size as usize),
+    )))
+}
+
+/// # Safety
+///
+/// This function should only be called with a QIR bigint created by the `__quantum__rt__bigint_*` functions.
+#[no_mangle]
+pub unsafe extern "C" fn __quantum__rt__bigint_get_data(input: *const BigInt) -> *const u8 {
+    ManuallyDrop::new((*input).to_signed_bytes_be()).as_ptr()
```

Comment:
That's the fun part, you don't! (Insert meme here...)
On a more serious note, the bigint `to_signed_bytes_be()` does not provide a view into the data but rather copies it into a new `Vec`. The QIR API spec doesn't describe a way to clean up the data returned by this function, which would imply that the assumption is that the pointer is a view into the bigint with library-managed lifetime. I don't know if there is a better way to support that with how Rust works here, or if we need updates to the QIR spec to capture the notion that this is a managed array. Perhaps it could return an opaque QirArray instead?
@bettinaheim for her thoughts on this one.